src/main/g8/src/test/scala/$package__packaged$/HelloWorldSpec.scala

```diff
@@ -0,0 +1,28 @@
+package $package$
+
+import org.http4s.dsl._
+import org.http4s.{Method, Request, Response, Status}
+import org.specs2.matcher.MatchResult
+
+class HelloWorldSpec extends org.specs2.mutable.Specification {
+  "HelloWorld" >> {
+    "return 200" >> {
+      uriReturns200()
+    }
+    "return hellow world" >> {
+      uriReturnsHelloWorld()
+    }
+  }
+
+  private[this] val retHelloWorld: Response = {
+    val getHW = Request(Method.GET, uri("/hello/world"))
+    val task = HelloWorld.service.run(getHW)
```

Comment:
There is an `service.orNotFound`, which gets rid of the `Option` so we don't need the `get`.  There are also some matchers in the http4s-testing module, which handle `Task`s.  (They're inspired by the old Scalaz Task matchers that were in earlier versions of Specs2).  That can avoid calls to `unsafeRun` in the specs.  (One is fine, but the second one, like line 27, is indicative of a missing flatMap.)

Reply:
Hmm, when I was looking around some of the older tests I saw a few tests that followed having two unsafeRuns. Also I had attempted to use `Http4sSpec` however it didnt look like it was an available library (or I just dont know the library to add to sbt).  I'll look into  refactoring that line to use a flatMap accordingly.

Reply:
Im not sure how to adapt that into a flatmap. once I have a `Response` I need to use unsafeRun to unwrap that to a string prior to the matching.  ie
`retHelloWorld.as[String].unsafeRun() ` the `as` method is resulting in another Task with an implicit EntityDecoder.  To get the result from that new Task im using another unsafeRun.

Do you have any bits of wisdom to point me in the right direction? I found https://raw.githubusercontent.com/http4s/http4s/master/testing/src/main/scala/org/http4s/testing/Http4sMatchers.scala and have been reading through that to see how it was done previously. But at the end of the day, I agree with you that having two unsafeRun's feels sloppy. Just not sure how to correct it

Reply:
You're right that many of the http4s tests still do multiple unsafeRuns.  Not everything has been cleaned up.  And so it's not a tragedy if we do.

If `retHelloWorld` returned a `Task[Response]` instead of a `Response`, that gets rid of one `unsafeRun`.  Then the individual tests could either:

- call a single `unsafeRun`
- use the matchers.  That would involve adding `"org.http4s" %% "http4s-testing" % version % "test"` to the library dependencies, extend `org.http4s.testing.Http4sMatchers`, and then could use things like `returnStatus` and `returnBody` in tests on the `Task` returned by `retHelloWorld`.